dash/development/_r_components_generation.py

```diff
@@ -639,22 +639,40 @@ def generate_exports(
     omitlist = ["utils.R", "internal.R"] + [
         "{}{}.R".format(prefix, component) for component in components
     ]
-    stripped_line = ""
     fnlist = []
 
     for script in os.listdir("R"):
         if script.endswith(".R") and script not in omitlist:
             rfilelist += [os.path.join("R", script)]
 
-    # in R, either = or <- may be used to create and assign objects
-    definitions = ["<-function", "=function"]
-
     for rfile in rfilelist:
         with open(rfile, "r") as script:
-            for line in script:
-                stripped_line = line.replace(" ", "").replace("\n", "")
-                if any(fndef in stripped_line for fndef in definitions):
-                    fnlist += set([re.split("<-|=", stripped_line)[0]])
+            # put the whole file on one line
```

Comment:
is this generic for loading R app file into a single line? 

 I can imagine a valid use case for testing using the same snippet.

we define the R app as a python triple quote string variable, and sanitize and convert the string to something we can run with `Rscript -e "{app}"`, @rpkyle  also points out other ways if we have the app defined in a R file.

Reply:
I made no attempt to ensure this code would still execute, in fact in general it will not. For that you would need to figure out which line breaks delineate statements and which don't, then separate the statements with semicolons and turn the mid-statement line breaks into spaces. That's a nontrivial task though; much better IMO would be to save the triple-quoted string to a temp file and run it from there.